lib/ansible/modules/system/pamd.py

```diff
@@ -507,20 +507,32 @@ def update_rule(self, rule_type, rule_control, rule_path,
         # Get a list of rules we want to change
         rules_to_find = self.get(rule_type, rule_control, rule_path)
 
+        changes = 0
         for current_rule in rules_to_find:
+            rule_changed = False
             if new_type:
-                current_rule.rule_type = new_type
+                if(current_rule.rule_type is not new_type):
+                    rule_changed = True
+                    current_rule.rule_type = new_type
             if new_control:
-                current_rule.rule_control = new_control
+                if(current_rule.rule_control is not new_control):
+                    rule_changed = True
+                    current_rule.rule_control = new_control
             if new_path:
-                current_rule.rule_path = new_path
+                if(current_rule.rule_path is not new_path):
+                    rule_changed = True
+                    current_rule.rule_path = new_path
             if new_args:
-                if isinstance(new_args, str):
-                    new_args = new_args.replace(" = ", "=")
-                    new_args = new_args.split(' ')
-                current_rule.rule_args = new_args
+                    if isinstance(new_args, str):
```

Comment:
Hmm this might be part of the problem. new_args is never a `str` at this point in the code because in the module spec it is defined as a list. So if you have in the task: 
`module_arguments: 'use_authtok,remember=6,retry=3'`

Then `new_args` at this point looks like: `['use_authtok remember=6 retry=3']` which is a list with a single item. We should either deprecate the form:
`module_arguments: 'use_authtok,remember=6,retry=3'`

and remove it from the examples section as a potential option OR we should handle parsing it into a list. Not sure how other Ansible modules handle this.